mne/preprocessing/nirs/tests/test_temporal_derivative_distribution_repair.py

```diff
@@ -23,18 +23,35 @@
 def test_temporal_derivative_distribution_repair(fname, tmp_path):
     """Test running artifact rejection."""
     raw = read_raw_nirx(fname)
-    raw = optical_density(raw)
+    raw_od = optical_density(raw)
+    raw_hb = beer_lambert_law(raw_od)
 
+    # With optical densities
     # Add a baseline shift artifact about half way through data
-    max_shift = np.max(np.diff(raw._data[0]))
+    max_shift = np.max(np.diff(raw_od._data[0]))
     shift_amp = 5 * max_shift
-    raw._data[0, 0:30] = raw._data[0, 0:30] - shift_amp
+    raw_od._data[0, 0:30] = raw_od._data[0, 0:30] - shift_amp
     # make one channel zero std
-    raw._data[1] = 0.
-    raw._data[2] = 1.
-    assert np.max(np.diff(raw._data[0])) > shift_amp
+    raw_od._data[1] = 0.
+    raw_od._data[2] = 1.
+    assert np.max(np.diff(raw_od._data[0])) > shift_amp
     # Ensure that applying the algorithm reduces the step change
-    raw = tddr(raw)
-    assert np.max(np.diff(raw._data[0])) < shift_amp
-    assert_allclose(raw._data[1], 0.)  # unchanged
-    assert_allclose(raw._data[2], 1.)  # unchanged
+    raw_od = tddr(raw_od)
+    assert np.max(np.diff(raw_od._data[0])) < shift_amp
+    assert_allclose(raw_od._data[1], 0.)  # unchanged
+    assert_allclose(raw_od._data[2], 1.)  # unchanged
+
+    # With Hb
+    # Add a baseline shift artifact about half way through data
+    max_shift = np.max(np.diff(raw_hb._data[0]))
+    shift_amp = 5 * max_shift
+    raw_hb._data[0, 0:30] = raw_hb._data[0, 0:30] - (1.1 * shift_amp)
```

Comment:
I guess maybe this could be reverted, too -- not sure if it's necessary or not

Reply:
@larsoner, yes that's the one I meant. Initially the tests wouldn't pass with just `- shift_amp` which I believe to be related to the fact that `raw_hb._data[0, 29] > raw_hb._data[0, 30]` which makes that when you subtract *shift_amp* to *raw_hb._data[0, 0:30]* we have `raw_hb._data[0, 30] - raw_hb._data[0, 29] < shift_amp`, so `assert np.max(np.diff(raw_hb._data[0])) > shift_amp` fails.

Reply:
Just tried and it fails without `1.1 *` because of the issue mentioned above. So all good with the tests from this PR I think.